docs/getting-started.rst

```diff
@@ -0,0 +1,308 @@
+zero-vector), but we'll change that in a moment. For now, go ahead and run the
+program a few times, changing the parameters to Vector and the speed and see
+what happens.
+
+Now that you're comfortable with the base mechanics of our new class, let's wire
+up our controls.
+
+First, we're going to define the four arrow keys as our controls. These can be
+set as class variables so we can change them later:
+
+``main.py``::
+
+   class Player(ppb.BaseSprite):
+       position = ppb.Vector(0, -3)
+       direction = ppb.Vector(0, 0)
+       speed = 4
+       left = keycodes.Left
+       right = keycodes.Right
+
+The ``keycodes`` module contains all of the keys on a US based keyboard. If you
+want different controls, you can look at the module documentation to find ones
+you prefer.
+
+Now, under our ``on_update`` function we're going to add two new event handlers.
+The snippet below doesn't include the class attributes we just defined, but
+don't worry, just add the new methods at the end of the class, beneath your
+``on_update`` method.
+
+``main.py``::
+
+   class Player(ppb.BaseSprite):
+
+
+       def on_key_pressed(self, key_event: KeyPressed, signal):
+           if key_event.key == self.left:
+               self.direction += ppb.Vector(-1, 0)
+           elif key_event.key == self.right:
+               self.direction += ppb.Vector(1, 0)
+
+       def on_key_released(self, key_event: KeyReleased, signal):
+           if key_event.key == self.left:
+               self.direction += ppb.Vector(1, 0)
+           elif key_event.key == self.right:
+               self.direction += ppb.Vector(-1, 0)
+
+So now, you should be able to move your player back and forth using the arrow
+keys.
+
+Reaching Out
+------------
+
+The next step will to make our player "shoot". I use shoot loosely here,
+your character can be throwing things, or blowing kisses, or anything, the only
+mechanic is we're going to have a new object start at the player, and fly up.
+
+First, we need a new class. We'll put it under ``Player``, but above ``setup``.
+
+
+``main.py``::
+
+   class Projectile(ppb.BaseSprite):
+      size = 0.25
+      direction = ppb.Vector(0, 1)
+      speed = 6
+
+      def on_update(self, update_event, signal):
+          if self.direction:
+              direction = self.direction.normalize()
+          else:
+              direction = self.direction
+          self.position += direction * self.speed * update_event.time_delta
+
+If we wanted to, we could pull out this ``on_update`` function into a mixin that
+we could use with either of these classes, but I'm going to leave that as an
+exercise to the reader. Just like the player, we can put a square image in the
+same folder with the name ``projectile.png`` and it'll get rendered, or we can
+let the engine make a colored square for us.
+
+Let's go back to our player class. We're going to add a new button to the class
+attributes, then update the ``on_key_pressed`` method. Just like before, I've
```

Comment:
Should clarify that we're not adding a new method, just adding code to the existing method.

Reply:
"update" seems find for that?